live-build/config/hooks/vm-artifacts/85-clear-etc-hostname.binary

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash -eux
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Set /etc/hostname to an empty string to allow transient hostnames to be set
+# by systemd-hostnamed. systemd-hostnamed gives precedence to static hostnames
+# if set. Since live-build sets a static hostname, we need to clear it to allow
+# transient hostnames to be set.
+#
+chroot binary truncate -s 0 /etc/hostname
```

Comment:
a bit late here, could we remove the file as another way to solve it too? I'm fine with this, just curious if we tried that..?